projects/rero/ng-core/src/lib/record/editor/services/editor.service.ts

```diff
@@ -94,9 +94,48 @@ export class EditorService {
    * @param field - FormlyFieldConfig, form config to be removed
    */
   removeHiddenField(field: FormlyFieldConfig) {
-    if (this._hiddenFields.some(f => f === field) && this.isFieldRoot(field)) {
+    if (this._hiddenFields.some(f => f === field) && this.isRoot(field.parent)) {
       this._hiddenFields = this._hiddenFields.filter(f => f.id !== field.id);
       this._hiddenFieldsSubject.next(this._hiddenFields);
     }
   }
+
+  /**
+   * Can the field be hidden?
+   * @param field - FormlyFieldConfig, the field to hide
+   * @returns boolean, true if the field can be hidden
+   */
+  canHide(field: FormlyFieldConfig) {
+    if (!field.templateOptions.longMode) {
+      return false;
+    }
+    return (
+      !field.templateOptions.required &&
+      !field.hide &&
+      field.hideExpression == null
+    );
```

Comment:
I like (me and myself) ternary instruction. In this case 
```
return (!field.templateOptions.longMode)
    ? false
    : ( !field.templateOptions.required &&
        !field.hide &&
        field.hideExpression == null
      );
```

Reply:
Seem to me more difficult to read. @sebastiendeleze any advises?

Reply:
+1 for use ternary, but only if expression is quite simple. In this case, it is difficult to read in my opinion. But I would suggest to merge statements like this:
```
return (
      field.templateOptions.longMode &&
      !field.templateOptions.required &&
      !field.hide &&
      field.hideExpression == null
    );
```